app/evm_keeper.go

```diff
@@ -0,0 +1,85 @@
+package app
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+// Convertion between native gas KAVA (6) to EVM (18).
+// 12 decimal difference, so 1_000_000_000_000.
+var conversionMultiplier = sdk.NewInt(1_000_000_000_000)
+
+// EVMBankKeeper is a wrapper for bank keeper that converts between EVM and native decimals
+type EVMBankKeeper struct {
+	bankKeeper bankkeeper.Keeper
+}
+
+var _ evmtypes.BankKeeper = (*EVMBankKeeper)(nil)
+
+// NewEVMBankKeeper returns a wrapped bank keeper that converts between EVM and native decimals
+func NewEVMBankKeeper(bk bankkeeper.Keeper) EVMBankKeeper {
+	return EVMBankKeeper{
+		bankKeeper: bk,
+	}
+}
+
+func (bk EVMBankKeeper) GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin {
+	bal := bk.bankKeeper.GetBalance(ctx, addr, denom)
+
+	return convertCoinToEvm(bal)
+}
+
+func (bk EVMBankKeeper) SendCoinsFromModuleToAccount(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error {
+	actualAmt := convertCoinsFromEvm(amt)
+
+	return bk.bankKeeper.SendCoinsFromModuleToAccount(ctx, senderModule, recipientAddr, actualAmt)
+}
+
+func (bk EVMBankKeeper) SendCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins) error {
+	actualAmt := convertCoinsFromEvm(amt)
+
+	return bk.bankKeeper.SendCoinsFromAccountToModule(ctx, senderAddr, recipientModule, actualAmt)
+}
+
+func (bk EVMBankKeeper) MintCoins(ctx sdk.Context, moduleName string, amt sdk.Coins) error {
+	actualAmt := convertCoinsFromEvm(amt)
+
+	return bk.bankKeeper.MintCoins(ctx, moduleName, actualAmt)
+}
+
+func (bk EVMBankKeeper) BurnCoins(ctx sdk.Context, moduleName string, amt sdk.Coins) error {
+	actualAmt := convertCoinsFromEvm(amt)
+
+	return bk.bankKeeper.BurnCoins(ctx, moduleName, actualAmt)
+}
+
+// convertCoinToEvm converts a sdk.Coin with native decimals to an sdk.Coin with evm decimals
+func convertCoinToEvm(coin sdk.Coin) sdk.Coin {
+	coin.Amount = coin.Amount.Mul(conversionMultiplier)
+
+	return coin
+}
+
+// convertCoinFromEvm converts an EVM sdk.Coin with 18 decimals back to native 6 decimal sdk.Coin
+func convertCoinFromEvm(coin sdk.Coin) sdk.Coin {
+	coin.Amount = coin.Amount.Quo(conversionMultiplier)
+
+	if coin.IsZero() {
```

Comment:
I *think* we actually want to truncate to zero here and return an error from any method converting from evm 18 decimals to 6 decimals.

Even though it's small, rounding up ends up operating on more coins than specified -- which should be avoided.  An explicit error is probably best?

Reply:
The edge cases also make great test examples -- most conversion code is much easier to understand from the tests in addition to having the implementation.

Reply:
Yeah, having an error here instead does makes sense. It should only be zero if it's dealing with values less than `1ukava` which shouldn't be valid anyways I suppose. Related thing I need to also check is if typical gas fees via wallets like MetaMask are less than `1ukava` which doesn't seem like great UX if users need to manually adjust it each time to not have it error.

Will add more tests for this after the weekend

Reply:
I think this might need specific logic to handle gas fees or some modification to fee calculations possibly in Ethermint, as some test transactions via MetaMask had default gas fees < 1ukava which causes transactions to fail with a panic.